src/components/navigation/AuthActionButton.tsx

```diff
@@ -5,29 +5,25 @@ import Button from '@material-ui/core/Button';
 import IconButton from '@material-ui/core/IconButton';
 import clsx from 'clsx';
 
-import { ModalCtx } from '@app/types';
-
-import { useModal } from '@app/contexts/ui/ModalProvider';
+import { useModal } from '@app/contexts/Modal';
```

Comment:
I would keep this in the ui directory. It's a ui context even if it doesn't have any ui in it.

Reply:
I don't think we'll need to organize our contexts that much until we hit 10-12 distinct context providers, even if some are UI-specific and others aren't. Directory organization is purely subjective of course, but do you feel this should be done right away?

Reply:
I think if you're willing to do something at 10-12 files you should just do it to begin with.

Reply:
I don't think it's necessary to preemptively organize components like that, and at this point the majority of our contexts would go in a `ui/` folder anyway so why not just keep them one directory up? This may just be where we differ in organization preferences but I'd prefer to stray away from nesting things that conceptually aren't that different.

Reply:
leaving this unresolved though because there certainly could be more discussion on this

Reply:
My point was this: say we have 20 different contexts in that directory, would you still make the same argument of not separating them because they're all contexts at the end of the day? If not, then when you do make that change down the road it will bring along a refactor with it. So why not just start breaking them out now? I guess it's essentially the same argument of if you know you're going to have to do something down the road just do it now

Reply:
That same point could be made about every file in this project. Why not divvy up the top-level components in `src/components` further? Won't we eventually have like 20 top-level route components? Of course, but hypotheticals don't make sense here - we _don't_ have that many right now, and we _don't_ have 20, or 10, contexts, even if the old contexts were still there. Plus, there is no implicit refactor involved in waiting; Code will instantly and reliably update imports for us.

My answer to "why not just do this now" is the same reason I didn't think refactoring the other context utilities was profitable. If we try to address use cases that literally do not exist yet, we are guaranteed to introduce tech debt. The odds of creating the ideal solution for a problem we don't fully understand aren't just unlikely, it's utterly impossible. Waiting until the problem exists is both the most convenient and the most logical approach. Sure, that's pretty dramatic for organizing files, but the philosophy is the same. There isn't a reason to solve a problem that long before it exists.

Reply:
was going to there is no refactor involved because code can fix imports for us but this doesn't seem to be the case with import aliasing. Maybe there's a fix but for now, sure, check out https://github.com/resident51/resident51/pull/4/commits/736686c9facfd3dd9c3c1760e753283feb4a5df9